lib/middleman-hashicorp/releases.rb

```diff
@@ -29,4 +29,21 @@ def self.fetch(product, version)
       end
     end
   end
+
+  #
+  # Gets only the most recently released version data for a given product
+  #
+  def self.fetch_latest_version(product)
+    url = "#{RELEASES_URL}/#{product}/index.json"
+    res = JSON.parse(open(url).read,
+      create_additions: false,
+      symbolize_names: true,
+    )
+    versions = res[:versions]
+
+    # Releases are formatted as an object rather than an array in the JSON
+    # structure, so we need to convert to an array then sort by version to
+    # get the latest. We then return all info for that version.
+    versions[versions.keys.sort_by(&Gem::Version.method(:new)).last]
+  end
```

Comment:
Since there was already a method in the releases class that fetches all versions from `releases.hashicorp.com`, I figured this would be the best place to add a variation that fetches only the latest version. I made the method a bit more general purpose, as I could see it being used for other things. For any given slug on `releases.hashicorp.com`, when called, it will return the full release data for only the latest version.